### ISSUE TITLE
Don't add traceparent header to some/all AWS endpoints

### ISSUE THREAD

**Issue by watson** [elastic/apm-agent-nodejs#1055]
We recently stopped adding the `elastic-apm-traceparent` header to outgoing HTTP requests to `*.s3.amazonaws.com` hosts (see #952).

But if the user is using regional hosts like `s3.us-west-2.amazonaws.com`, this will still break. At a minimum, we should therefore also apply the same logic to hosts with the pattern `*.s3.*.amazonaws.com`.

But should strongly consider expanding the query to `*.amazonaws.com`. But before doing so we should consider if this will affect our ability to capture distributed traces with AWS Lambda functions somewhere in the trace (and potentially also other AWS services).

**Comment by watson** [comment]
Alternative approach: Look for the signature and don't add the traceparent header if it's present

**Comment by watson** [comment]
I investigated it a bit further and have a solution running locally that just looks at the signature header for version 4 of the AWS signature (supporting sig v2 should also be possible, but I'm not sure it's worth it).

But during the investigation, I realized that my fear about matching all `*.amazonaws.com` requests was probably unfound. Lambdas are not called directly with an HTTP request, so all invocations will go through the AWS API and as such should not have any of our headers added.

So maybe we should just look at the domain `*.amazonaws.com` after all 🤔
